src/types.rs

```diff
@@ -35,17 +35,26 @@ impl AccessToken {
     }
 }
 
-impl AsRef<str> for AccessToken {
+impl AsRef<str> for Token {
     fn as_ref(&self) -> &str {
         self.as_str()
     }
 }
 
-impl From<TokenInfo> for AccessToken {
+impl From<TokenInfo> for Token {
     fn from(value: TokenInfo) -> Self {
-        AccessToken {
-            value: value.access_token,
-            expires_at: value.expires_at,
+        match value {
+            TokenInfo {
+                access_token: Some(value),
+                expires_at,
+                ..
+            } => Token { value, expires_at },
+            TokenInfo {
+                id_token: Some(value),
+                expires_at,
+                ..
+            } => Token { value, expires_at },
+            _ => todo!(),
```

Comment:
Not sure how to best approach this. Changing `Token` further to expose the APIs to access specific (id, access) token types might help?